File: usedesk_sdk/src/main/java/ru/usedesk/sdk/internal/domain/repositories/knowledge/IKnowledgeBaseRepository.java
package ru.usedesk.sdk.internal.domain.repositories.knowledge;

import android.support.annotation.NonNull;

import java.util.List;

import ru.usedesk.sdk.external.entity.exceptions.DataNotFoundException;
import ru.usedesk.sdk.external.entity.exceptions.UsedeskHttpException;
import ru.usedesk.sdk.external.entity.knowledgebase.ArticleBody;
import ru.usedesk.sdk.external.entity.knowledgebase.ArticleInfo;
import ru.usedesk.sdk.external.entity.knowledgebase.Category;
import ru.usedesk.sdk.external.entity.knowledgebase.SearchQuery;
import ru.usedesk.sdk.external.entity.knowledgebase.Section;

public interface IKnowledgeBaseRepository {
    @NonNull
    List<Section> getSections(@NonNull String id, @NonNull String token) throws UsedeskHttpException;

    @NonNull
    ArticleBody getArticleBody(@NonNull String id, @NonNull String token, long articleId) throws UsedeskHttpException;

    @NonNull
    List<ArticleBody> getArticles(@NonNull String id, @NonNull String token,
                                  @NonNull SearchQuery searchQuery) throws UsedeskHttpException;

    @NonNull
    List<Category> getCategories(@NonNull String id, @NonNull String token, long sectionId)
            throws UsedeskHttpException, DataNotFoundException;

    @NonNull
    List<ArticleInfo> getArticles(@NonNull String id, @NonNull String token,
                                  long categoryId) throws UsedeskHttpException, DataNotFoundException;

    void addViews(@NonNull String accountId, @NonNull String token, long articleId)
            throws UsedeskHttpException, DataNotFoundException;
}

File: knowledgebase-sdk/src/main/java/ru/usedesk/knowledgebase_sdk/data/repository/IKnowledgeBaseRepository.kt
package ru.usedesk.knowledgebase_sdk.data.repository

import ru.usedesk.common_sdk.entity.exceptions.UsedeskDataNotFoundException
import ru.usedesk.common_sdk.entity.exceptions.UsedeskHttpException
import ru.usedesk.knowledgebase_sdk.data.repository.entity.UsedeskArticleBodyOld
import ru.usedesk.knowledgebase_sdk.data.repository.entity.UsedeskArticleInfoOld
import ru.usedesk.knowledgebase_sdk.data.repository.entity.UsedeskCategoryOld
import ru.usedesk.knowledgebase_sdk.data.repository.entity.UsedeskSectionOld
import ru.usedesk.knowledgebase_sdk.entity.*

internal interface IKnowledgeBaseRepository {
    @Throws(UsedeskHttpException::class)
    fun getSections(accountId: String, token: String): List<UsedeskSectionOld>

    @Throws(UsedeskHttpException::class)
    fun getArticleBody(accountId: String, token: String, articleId: Long): UsedeskArticleBodyOld

    @Throws(UsedeskHttpException::class)
    fun getArticles(accountId: String, token: String, searchQuery: UsedeskSearchQueryOld): List<UsedeskArticleBodyOld>

    @Throws(UsedeskHttpException::class, UsedeskDataNotFoundException::class)
    fun getCategories(accountId: String, token: String, sectionId: Long): List<UsedeskCategoryOld>

    @Throws(UsedeskHttpException::class, UsedeskDataNotFoundException::class)
    fun getArticles(accountId: String, token: String, categoryId: Long): List<UsedeskArticleInfoOld>

    @Throws(UsedeskHttpException::class, UsedeskDataNotFoundException::class)
    fun addViews(accountId: String, token: String, articleId: Long)
}